src/app/plugins/modreport.plugin.ts

```diff
@@ -10,22 +10,25 @@ export class ModReportPlugin extends Plugin {
   public pluginAlias = [];
   public permission: ChannelType = ChannelType.Staff;
   public pluginChannelName: string = Constants.Channels.Staff.UserOffenses;
-
-  private _commandPattern: RegExp = /(add|list|warn|ban)\s+([^#]+#\d{4})\s*(.*)/;
+  public _commandPattern: RegExp = /(add|list|warn|ban)\s+([^#]+#\d{4})\s*(.*)/;
 
   constructor(public container: IContainer) {
     super();
   }
 
-  public async execute(message: IMessage, args: string[]) {
-    const match_arr = args.join(' ').match(this._commandPattern);
+  public validate(message: IMessage, args: string[]) {
+    if (!args.length) return false;
```

Comment:
why didn't you just do the following?
`
const match_arr = args.join(' ').match(this._commandPattern)
return !!match_arr
`

Reply:
Also, is there a reason you overwrite the validate function for this one despite it looking like it does the same thing?

Reply:
Already changed that in another commit